### ISSUE TITLE
Expert Mode off, blocks certain quests from being activated

### ISSUE THREAD

**Issue by RubenKelevra** [Helium314/StreetComplete#612]
Currently certain quests cannot be activated if the switch "Expert Mode" in the settings is not on.

Given that this is Street Complete Expert Edition, I'm not sure it needs a switch to turn on "Expert Mode".

But this aside, since those quests are shown like any other quest, but just not react to turning them on - without any further hint why this is the case, the UI/functionality does not follow the Principle Of Least Astonishment.

I recommend either
- removing the Expert Switch entirely
- removing the functionality that it blocks certain quests
- add a popup if the user tries to enable those disabled quests
- remove those quests from being listed, if expert mode is not on.

**How to Reproduce**
See https://github.com/Helium314/SCEE/issues/611

**Expected Behavior**
The UI should follow the Principle Of Least Astonishment.

**Does it happen in normal StreetComplete?**
Nope

**Versions affected**
Android 12
SCEE [58.21](https://github.com/Helium314/SCEE/releases/tag/v58.21) and [59.0-alpha1](https://github.com/Helium314/SCEE/releases/tag/v59.0-alpha1)

**Comment by Helium314** [comment]
> Given that this is Street Complete Expert Edition, I'm not sure it needs a switch to turn on "Expert Mode".

In my opinion it does. If you check notes created with SCEE, you will often notice users that are pretty new to OSM creating notes that show those are not secondary accounts of experienced users.
Those who don't even go to settings essentially will have SC, and those who do will see an explicit warning with the unexpected "understood" instead of "ok" button before they can use the more dangerous capabilities of SCEE.

**Comment by mnalis** [comment]
> add a popup if the user tries to enable those disabled quests

... is the only one that would improve situation, IMHO.  

Some message there (e.g. _"Those quests cannot be enabled unless expert mode is enabled, because they might require intimate knowledge of OSM workings in order not to damage data or otherwise produce unwanted results"_ or something similarly scary) when clicking on disabled quest might be helpful to explain why those are disabled...

**Comment by RubenKelevra** [comment]
> > Given that this is Street Complete Expert Edition, I'm not sure it needs a switch to turn on "Expert Mode".
> 
> In my opinion it does. If you check notes created with SCEE, you will often notice users that are pretty new to OSM creating notes that show those are not secondary accounts of experienced users. Those who don't even go to settings essentially will have SC, and those who do will see an explicit warning with the unexpected "understood" instead of "ok" button before they can use the more dangerous capabilities of SCEE.

I might not have been clear earlier: I actually like that you can toggle advanced editing features like tags. I was just wondering if labeling it ‘Expert Mode’ when you’ve already installed the expert edition is necessary. It feels a bit redundant.

**Comment by RubenKelevra** [comment]
> > add a popup if the user tries to enable those disabled quests
> 
> ... is the only one that would improve situation, IMHO.
> 
> Some message there (e.g. _"Those quests cannot be enabled unless expert mode is enabled, because they might require intimate knowledge of OSM workings in order not to damage data or otherwise produce unwanted results"_ or something similarly scary) when clicking on disabled quest might be helpful to explain why those are disabled...

Yeah that's my favorite solution, too.

**Comment by HolgerJeromin** [comment]
> I was just wondering if labeling it ‘Expert Mode’ when you’ve already installed the expert edition is necessary. It feels a bit redundant.

We are the experts of the experts :-D

**Comment by RubenKelevra** [comment]
I noticed that every single quest which is part of the expert edition got a prompt of some kind if you activate it.

Maybe we could remove them? 🤔

I mean the user already activated that he/she wants to use the expert features, I don't think there's a need to remind the user on every quest they activate that they choose to do so.

**Comment by rusty-snake** [comment]
Some have just the default message, but others show additional information why it is an expert quest. I find this very useful.

**Comment by RubenKelevra** [comment]
I don't. It's just annoying:


https://github.com/user-attachments/assets/efd38f2b-373c-46a6-8cff-ffddb99604de

**Comment by rusty-snake** [comment]
Your usecase it to enable all SCEE quests?

**Comment by RubenKelevra** [comment]
Yeah, hat to reinstall an older version to continue to map, as the Alpha is currently unusable.

But that's an issue on every new profile for Quests you create. It's so annoying that I usually clone the "all" preset and then rather disable than enable quests.

Its such a nuisance.

There's *zero* advantage of having all those pop-ups, it just makes the app harder to use.

**Comment by rusty-snake** [comment]
Well I enable only selected SCEE quest where I want to contribute data, have knowledge in the tagging of this topic. Having all enabled is just to spammy in my area and I don't want to mess-up the database because I forgot that that "foo" is a SCEE quest that can only be answered on 70% of the objects it shows on.

**Comment by RubenKelevra** [comment]
And? 🤔

I still don't see an advantage of annoy user with useless pop-ups.

**Comment by mcliquid** [comment]
I can understand the frustration, but the question is, how often does this happen? In my opinion, only the very first time. And then you have to assume that the user has never used SCEE before. In that case, the pop-ups are necessary to draw attention to the fact that the individual quests are not easy to answer and should be chosen carefully.

As soon as you have activated the desired quests after the first installation, the problem no longer occurs. You can make gradations in the individual presets and export everything as a backup at the end in case you have to reinstall the app.

**Comment by rusty-snake** [comment]
- Warning them, reminding them against that they use SC*EE*.
  - Did you really read the wiki for the tagging schema for all of them? Railway, SAC, ... are different topics. You may not know that much about them all. I find some SC quest already hard to answer correct w/o knowing the wiki.
- Providing useful information in some cases.
  - For some quests (e.g. parking orientation) it does not seem to provide much benefit. Rather easy quest, no additional information.
  - For some quest that expect you to edit the query I want to know that.

After all I'm not the person how decides this. I've said that I would like to keep them. No reasons to further discuss this. If there is someone how you need to convince it's not me.

**Comment by michaelblyons** [comment]
You already have a way to show "disabled by default" and "disabled in {your region}" which can both be shown for the same quest. Please also add a "disabled without expert mode" (wording could be better) one, too

**Comment by Helium314** [comment]
> You already have a way to show "disabled by default" and "disabled in {your region}" which can both be shown for the same quest. Please also add a "disabled without expert mode" (wording could be better) one, too

That's the only suggestion so far that should be simple to implement and not have unwanted side effects.

**Comment by Korb** [comment]
> add a popup if the user tries to enable those disabled quests

If a user sees a blocked list item, they're unlikely to click it and, therefore, won't see a notification about why the quest is blocked.

I often see the "Pro" label on websites and proprietary apps when certain features are unavailable with the current plan ([screenshots for examples](https://drive.google.com/drive/folders/15infSZtWUF_ioGLGuHGLCSnVGK7Z4G9_)). You can use this same interface trick, familiar to most users, by adding "Expert" to the blocked item. Or by replacing the checkbox with "Expert." This will eliminate the confusion I also found myself in, not understanding why some quests were unavailable and not having any notifications or warnings about it.
